src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -889,59 +889,35 @@ private AttributeListSyntax ParseAttributeDeclaration()
                 ? _syntaxFactory.AttributeTargetSpecifier(ConvertToKeyword(this.EatToken()), this.EatToken(SyntaxKind.ColonToken))
                 : null;
 
-            var attributes = _pool.AllocateSeparated<AttributeSyntax>();
-            this.ParseAttributes(attributes);
+            var attributes = this.ParseCommaSeparatedSyntaxList(
+                ref openBracket,
+                SyntaxKind.CloseBracketToken,
+                static @this => @this.IsPossibleAttribute(),
+                static @this => @this.ParseAttribute(),
+                static (@this, openBracket, list, _, _) => (openBracket, skipBadAttributeListTokens(@this, list)),
```

Comment:
note: Teh lack of passing along the expected/closekinds are also strange cases where teh list parsing deviates.  i'd like to always consistently pass them along.  but i don't feel super strongly about it.

Reply:
going to try that out in another branch.